shell/platform/android/io/flutter/view/ResourceExtractor.java

```diff
@@ -281,8 +290,13 @@ private boolean extractUpdate(File dataDir) {
             try {
                 if (entry.getName().endsWith(".bzdiff40")) {
                     ByteArrayOutputStream diff = new ByteArrayOutputStream();
-                    try (InputStream is = zipFile.getInputStream(entry)) {
+                    InputStream is = zipFile.getInputStream(entry);
```

Comment:
Did you mean for this to be part of this PR?

Reply:
Yeah, though I'm not set on it.

Try-with-resources is Java 8 only, so it was failing the lint. But you can build apps for older SDKs that use some Java 8 features (including this one) https://developer.android.com/studio/write/java8-support.html, and we do already have this set up in `flutter.gradle`. So practically I think this is fine given how the engine code is actually used, even though there's no way to guarantee it here. Would it be better to suppress this type of warning instead? That's probably cleaner than writing around all Java 8 APIs.

Reply:
Part of the reason to upgrade the engine to API 28 was to enable Java 8 features in it.  We might be able to fix this by just updating our AndroidManifest.xml to say we're targetting API 28 - right now it has something like 21.  Can you try with that?

Reply:
I did try giving that a go when I first worked on this, and unfortunately that wasn't taken into account for the lint. I think it's probably because supporting Java 8 on earlier SDKs requires a specific Android Gradle plugin and build config to work properly. Just the `targetSdkVersion` in the manifest on its own doesn't necessarily mean that it's going to be compiled in a backwards compatible way from what I can tell.

Reply:
Maybe this is a lint we can turn off, as it would be caught at compile time otherwise for people.  But maybe we should have a broader discussion about the priority of having Java 8 features available in the embedder.

Reply:
I think the way we use it it's likely safe. I'll revert these changes and see if I can turn off the lint for just this specific Java 8 usage.